flux/src/test/java/com/groupbyinc/util/UrlFunctionsBugTest.java

```diff
@@ -0,0 +1,118 @@
+                url);
+    }
+
+    @Test
+    public void splitTest() throws IOException {
+
+        String refinementGood = "~gender=Women~simpleColorDesc=Pink~product=Clothing";
+
+        String[] split = refinementGood.split(REFINEMENT_REGEX);
+
+        assertArrayEquals(new String[]{"", "gender=Women", "simpleColorDesc=Pink", "product=Clothing"}, split);
+
+        String refinementBad = "~category_leaf_expanded=Category Root~Athletics~Men's~Sneakers";
+
+        split = refinementBad.split(REFINEMENT_REGEX);
+
+        assertArrayEquals(
+                new String[]{
+                        "", "category_leaf_expanded=Category Root~Athletics~Men's~Sneakers"}, split);
+
+        String refinementWorst =
+                "~category_leaf_expanded=Category Root~Athletics~Men's~Sneakers~category_leaf_id=580003";
+
+        split = refinementWorst.split(REFINEMENT_REGEX);
+
+        assertArrayEquals(
+                new String[]{
+                        "", "category_leaf_expanded=Category Root~Athletics~Men's~Sneakers", "category_leaf_id=580003"},
+                split);
+
+        split = "test=bob~price:10..20".split(REFINEMENT_REGEX);
+        assertArrayEquals(new String[]{"test=bob", "price:10..20"}, split);
+
+
+        split = ("test=bob~price:10..20~category_leaf_expanded=Category Root~Athletics~Men's" +
+                 "~Sneakers~category_leaf_id=580003~color=BLUE~color=YELLOW~color=GREY").split(REFINEMENT_REGEX);
+        assertArrayEquals(new String[]{"test=bob", "price:10..20",
+                                       "category_leaf_expanded=Category Root~Athletics~Men's~Sneakers", "category_leaf_id=580003",
+                                       "color=BLUE", "color=YELLOW", "color=GREY"}, split);
+
+        String complexString =
+                "~category_leaf_expanded=Category Root~Athletics~Men's~Sneakers~category_leaf_id=580003~color=BLUE~color=YELLOW~color=GREY";
+
+        split = complexString.split(REFINEMENT_REGEX);
+
+        assertArrayEquals(
+                new String[]{
+                        "", "category_leaf_expanded=Category Root~Athletics~Men's~Sneakers", "category_leaf_id=580003",
+                        "color=BLUE", "color=YELLOW", "color=GREY"
+                },
+                split);
+
+        String reallyLongString =
+                "~category_leaf_expanded=Category Root~Athletics~Men's~Sneakers~category_leaf_id=580003~" +
+                "color=BLUE~color=YELLOW~color=GREY~feature=Lace Up~feature=Light Weight~brand=Nike";
+
+        split = reallyLongString.split(REFINEMENT_REGEX);
+
+        assertArrayEquals(
+                new String[]{
+                        "", "category_leaf_expanded=Category Root~Athletics~Men's~Sneakers", "category_leaf_id=580003",
+                        "color=BLUE", "color=YELLOW", "color=GREY", "feature=Lace Up", "feature=Light Weight",
+                        "brand=Nike"
+                },
+                split);
+    }
+}
```

Comment:
You need tests that test the edge conditions for the method call that you're now going to create.

```java
testNull()
testEmpty()
testUtf8()
```